layers/+tools/pdf-tools/README.org

```diff
@@ -0,0 +1,192 @@
+ - [[Install][Install]]
+   - [[Prerequisites][Prerequisites]]
+   - [[Dotfile][Dotfile]]
+ - [[Usage][Usage]]
+ - [[Known Issues ][Known Issues ]]
+ - [[Key bindings][Key bindings]]
+   - [[PDF View][PDF View]]
+   - [[Leader keys][Leader keys]]
+   - [[Outline buffer mode][Outline buffer mode]]
+   - [[Annotation list mode][Annotation list mode]]
+   - [[Occur mode][Occur mode]]
+
+* Description
+   This layer enables support for the [[https://github.com/politza/pdf-tools][=pdf-tools=]] package.
+   
+According to the official repository, "PDF Tools is, among other things, a
+replacement of DocView for PDF files. The key difference is, that pages are not
+pre-rendered by e.g. ghostscript and stored in the file-system, but rather
+created on-demand and stored in memory."
+
+Examples of features that =pdf-tools= provides:
+- Use =occur= in the pdf file;
+- Show all headings in a outline buffer;
+- Manipulate annotations.
+
+* Install
+
+** Prerequisites
+Linux is the only operating system officially supported, but it's possible to
+use =pdf-tools= on OS X as well, and possibly on other Unix flavors.
+
+You'll need to install a few libraries. Check the [[https://github.com/politza/pdf-tools#server-prerequisites][instructions]] on
+=pdf-tools= page. Note that compiling from source might not be necessary, as
+the documentation states.
+
+If you use a Debian based system such as Ubuntu, you can skip this step, as
+=pdf-tools= installation asks whether you want it to try to download
+automatically all dependencies using ~apt-get~.
+
+
+** Dotfile
+After installing the dependencies, add this to your =~/.spacemacs=.
+
+#+begin_src emacs-lisp
+  (setq-default dotspacemacs-configuration-layers '(pdf-tools))
+#+end_src
+
+After that, syncronise your configuration with ~SPC f e R~. This will pop up a
+window showing the compilation process.
+
+If everything goes right, the compilation won't display any error and you will
+be able to see /PDFView/ (instead of /DocView/) on the spaceline after opening a
+pdf file.
+
+If the compilation run successfully but you get 
+~Error (use-package): pdf-tools :config: No executable `epdfinfo' found~,
+then just restart Spacemacs ([[https://github.com/politza/pdf-tools/issues/31][issue]]).
+
+If the compilation does not run successfully, you should confirm whether all
+dependencies are properly installed and then run the command ~pdf-tools-install~.
+
+* Usage
+Try out opening the outline (~o~). The outline window shows the structure of the
+pdf, displaying all headings.
+
+You can set annotations by selecting regions and right clicking on the buffer.
+Additionally, all annotation related functions are can be accessed with ~SPC m a~ or
+~M-m a~.
+
+Searching with =occur= can be done by using =pdf-occur= (~SPC m s o~ or ~M-m m o~).
+
+Tips:
+- You'll probably want to use =pdf-view-set-slice-from-bounding-box= (bounded to
+  ~SPC m s b~ and ~M-m s b~.) This command trims the blank parts of the
```

Comment:
`M-m` is the leader key-chord in the emacs editing style so what @nixmaniack said